writer.go

```diff
@@ -223,7 +226,10 @@ func (z *Writer) Write(buf []byte) (int, error) {
 }
 
 // compressBlock compresses a block.
-func (z *Writer) compressBlock(data []byte) error {
+func (z *Writer) compressBlock(dataBlock []byte) error {
+	data := make([]byte, len(dataBlock))
```

Comment:
Move it up before this is called and use the getBuffer function call,.

Reply:
oh, I missed this comment, it seems we might be able to use those pools that are already defined

Reply:
I have updated it, I am new to the code and didn't know there were already pools defined for each block size :)